gomint-api/src/main/java/io/gomint/event/world/SignChangeTextEvent.java

```diff
@@ -48,39 +61,58 @@ public Block getBlock() {
      *
      * @param line    which should be set
      * @param content which should be set on that line
+     * @deprecated Use {@link #setLines(List)} instead
      */
-    public void setLine( int line, String content ) {
+    @Deprecated
+    public void setLine(int line, String content) {
         // Silently fail when line is incorrect
-        if ( line > 4 || line < 1 ) {
+        if (line > 4 || line < 1) {
             return;
         }
 
-        if ( this.lines.size() < line ) {
-            for ( int i = 0; i < line - this.lines.size(); i++ ) {
-                this.lines.add( "" );
+        if (this.lines.size() < line) {
+            for (int i = 0; i < line - this.lines.size(); i++) {
+                this.lines.add("");
             }
         }
 
-        this.lines.set( line - 1, content );
+        this.lines.set(line - 1, content);
     }
 
     /**
      * Get a specific line of the sign content
      *
      * @param line which you want to get
-     * @return string or null when not setö
+     * @return string or null when not set
      */
-    public String getLine( int line ) {
+    public String getLine(int line) {
         // Silently fail when line is incorrect
-        if ( line > 4 || line < 1 ) {
+        if (line > 4 || line < 1) {
             return null;
         }
 
-        if ( this.lines.size() < line ) {
+        if (this.lines.size() < line) {
             return null;
         }
 
-        return this.lines.get( line - 1 );
+        return this.lines.get(line - 1);
+    }
+
+    /**
+     * Get all lines which the sign previously had
+     *
+     * @return all lines which the sign previously had
+     */
+    public List<String> getLines() {
+        return lines;
     }
 
+    /**
+     * Set the lines the sign should have after the event call
+     *
+     * @param lines the lines the sign should have
+     */
+    public void setLines(List<String> lines) {
+        this.lines = lines;
```

Comment:
This should not be able to replace the internal list implementation

Reply:
I will add a check here if the amount of lines is greater than 3 but it's not unsafe to set the list directly

Reply:
I might cause issues when one eventhandler sets this to an Unmodifiable list for example, and then an other eventhandler tries to modify the list. But it should cause no issues internally.

Reply:
Yeah get and size should be checked for. Also on getLines

Reply:
Resolved by 557b821f1e2c126ac96c0f3757f891d6f15dc512